jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/PluginConfigurationProcessor.java

```diff
@@ -51,6 +62,12 @@
   static AbsoluteUnixPath getAppRootChecked(JibPluginConfiguration jibPluginConfiguration)
       throws MojoExecutionException {
     String appRoot = jibPluginConfiguration.getAppRoot();
+    if (appRoot.isEmpty()) {
```

Comment:
Looking at this again, the way we "compute" the app root and the base image is different from Gradle now. For Gradle, we did not touch its `getAppRootChecked()` or add `getBaseImage()`. Instead, [we explicitly updated the actual parameters (somewhat like lazily and dynamically setting a "default" app root value](https://github.com/GoogleContainerTools/jib/pull/923/files#diff-aef8bf567c91df6622aa0ea29ed0c682R133)) (hence anyone calling `ContainerParameters.getAppRoot()` returns the lazily set default). I'd like to be consistent.

Either way makes sense in some way:
1. lazily setting the default: conceptually the parameter is set to some default. It's just that it's bound dynamically. (However, until the default value is set later, the parameter value is like in an undefined state.)
1. leaving the parameter empty: no value is actually provided by the user, and the default is just an empty string or null (although this default is an invalid value semantically). It's just that we notice this, "compute" a reasonable value, and pass the computed value to various components.

@coollog WDYT? At least, we need to be consistent with Gradle.

Reply:
Okay, sounds good. The current PR is good, and Let's move Gradle to option 2 later.